crates/net/network/src/cache.rs

```diff
@@ -48,13 +48,29 @@ impl<T: Hash + Eq + fmt::Debug> LruCache<T> {
         (new, evicted)
     }
 
+    /// Gets the given element, if exists, and promotes to lru.
+    pub fn get(&mut self, entry: &T) -> Option<&T> {
+        let _ = self.inner.get(entry)?;
+        self.iter().next()
```

Comment:
this is the same as:
```suggestion
        let _ = self.inner.get(entry)?;
        Some(entry)
```
?

Reply:
not if the types have a custom `PartialEq` impl, consider type
```rust
struct MyType {
    id: i32,
    colour: i32,
}

impl PartialEq for MyType {
    fn eq(&self, other: &Self) -> bool {
        self.id == other.id
    }
}

impl Hash ..
```
then we can have two instances with different colour, that will both have the same key
```rust
let green = MyType { id: 1, colour: 32 };
let blue = MyType { id: 1, colour: 34 };

let mut cache = LruCache::new();
cache.insert(green);
let _ = cache.get(&blue); // this will exist, but it will actually be green that is in the map
```
this extension to LruCache just makes it safe to use in the codebase kind of, cause we have a couple of types that impl custom `PartialEq` on a subset of fields. using a `HashMap` for these types makes way more sense though.